src/facilities-context/FacilitiesContext.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { useEffect } from "react";
+
+import { Facility } from "../page-multi-facility/types";
+import useScenario from "../scenario-context/useScenario";
+import { facilitiesReducer } from "./reducer";
+import * as actions from "./actions";
+
+export type FacilityMapping = { [key in Facility["id"]]: Facility }
+
+export interface FacilitiesState {
+  loading: boolean;
+  failed: boolean;
+  facilities: FacilityMapping;
+}
+
+export type FacilitiesDispatch = (action: actions.FacilitiesActions) => void;
+
+export type ExportedAsyncActions = {
+  updateFacility: (...props: any) => Promise<void>;
+  removeFacility: (...props: any) => Promise<void>;
+  // fetchRtData: ExportedAsyncFunction;
+}
+
+export interface FacilitiesContext {
+  state: FacilitiesState;
+  dispatch: FacilitiesDispatch | undefined;
+  asyncActions: ExportedAsyncActions;
+}
+
+export const FacilitiesContext = React.createContext<FacilitiesContext | undefined>(
+  undefined,
+);
+
+export const FacilitiesProvider: React.FC<{ children: React.ReactNode }> = ({
+  children,
+}) => {
+  const [state, dispatch] = React.useReducer(facilitiesReducer, {
+    loading: true,
+    failed: false,
+    facilities: {},
+  });
+
+  const [scenario] = useScenario();
+  const scenarioId = scenario?.data?.id
+
+  useEffect(() => {
+    if (scenarioId) {
+      actions.fetchFacilities(scenarioId, dispatch);
+    }
+  }, [scenarioId]);
+
+  const asyncActions = {
```

Comment:
My intention with these `asyncActions` was to keep the database calls in one place (i.e. the `actions.tsx` file), and then export that from the state provider as an available function to retrieve from `useFacilities`. Then it could be used in the components like this and the app state dispatching will happen in the action: 

```
updateFacility(scenarioId, facilityUpdates);

deleteFacility(scenarioId, facilityId);

fetchRtData(facility);
```

Reply:
this is pretty sweet